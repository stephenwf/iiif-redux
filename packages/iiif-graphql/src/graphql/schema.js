import { makeExecutableSchema } from 'graphql-tools';

import typeDefs from '../schema.graphql';
import resolvers from '../resolvers';
import schemaDirectives from './directives';

const schema = makeExecutableSchema({
  typeDefs,
  resolvers,
  schemaDirectives,
});

export default schema;
