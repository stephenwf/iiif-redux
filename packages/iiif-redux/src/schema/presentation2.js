/**
 * Presentation 2 - schema definition
 * ============================================================================
 *
 * - Collection
 * - Manifest
 * - Sequence
 * - Canvas
 * - Annotation
 * - AnnotationList
 * - Range
 * - Layer
 * - Image Content (Annotation)
 * - Other Content (Annotation List)
 *
 * - External resources
 * - Services
 */

// @todo Image Service API.
import { compose } from 'redux';
import { schema, normalize } from 'normalizr';
import moveStartCanvasToSequence from '../compat/moveStartCanvasToSequence';
import preprocessLinkedEntities from '../compat/preprocessLinkedEntities';
import addMissingIds from '../compat/addMissingIds';

function createEntity(name) {
  const options = {
    idAttribute: '@id',
  };

  return new schema.Entity(name, {}, options);
}

const collection = createEntity('collections');
const manifest = createEntity('manifests');
const sequence = createEntity('sequences');
const canvas = createEntity('canvases');
const annotation = createEntity('annotations');
const annotationList = createEntity('annotationLists');
const range = createEntity('ranges');
const layer = createEntity('layers');
const imageResource = createEntity('imageResources'); // thumbnails + image service
const choice = createEntity('choices');

// Unofficial types?
const externalResource = createEntity('externalResources');
const service = createEntity('services', true);
const within = new schema.Array(
  {
    collection,
    manifest,
    externalResource,
    layer,
  },
  entity => {
    switch (entity['@type']) {
      case 'sc:Collection':
        return 'collection';
      case 'sc:Manifest':
        return 'manifest';
      case 'sc:Layer':
        return 'layer';
      default:
        return 'externalResource';
    }
  }
);

const rangeMember = new schema.Union(
  {
    canvas,
    range,
  },
  input => (input['@type'] === 'sc:Canvas' ? 'canvas' : 'range')
);

/**
 * Dereferencable resources:
 *
 * - Collection (Required)
 * - Manifest (Required)
 * - First Sequence (Optional)
 * - Nth Sequence (Required)
 * - Canvas (Optional)
 * - Annotation (Recommended)
 * - AnnotationList (Required)
 * - Range (Optional)
 * - Layer (Optional)
 * - Image Content (Required)
 * - Other Content (Required)
 */
const RESOURCE_TYPE_MAP = {
  'sc:Collection': 'collection',
  'sc:Sequence': 'sequence',
  'sc:Manifest': 'manifest',
  'sc:Canvas': 'canvas',
  'sc:AnnotationList': 'annotationList',
  'sc:Annotation': 'annotation',
  'sc:Range': 'range',
  'sc:Layer': 'layer',
  'dctypes:Image': 'imageResource',
};
const resource = new schema.Union(
  {
    collection,
    sequence,
    manifest,
    canvas,
    annotationList,
    annotation,
    range,
    layer,
    imageResource,
    service,
  },
  entity => {
    if (RESOURCE_TYPE_MAP[entity['@type']]) {
      return RESOURCE_TYPE_MAP[entity['@type']];
    }
    if (entity.profile) {
      return 'service';
    }
    throw Error('Unknown entity type');
  }
);

// ===========================================================================
// 1) Collections
// ===========================================================================
collection.define({
  collections: new schema.Array(
    {
      collection,
      manifest,
    },
    entity => (entity['@type'] === 'sc:Manifest' ? 'manifest' : 'collection')
  ),
  manifests: [manifest],
  members: new schema.Array(
    {
      collection,
      manifest,
    },
    entity => (entity['@type'] === 'sc:Manifest' ? 'manifest' : 'collection')
  ),

  // Linking
  seeAlso: [externalResource],
  service: [service],
  related: [externalResource],
  rendering: [externalResource],
  otherContent: [annotationList],
  within: within,

  // Extra
  thumbnail: imageResource,
});

// ===========================================================================
// 2) Manifests
// ===========================================================================
manifest.define({
  sequences: [sequence],
  structures: [range],

  // Linking
  seeAlso: [externalResource],
  service: [service],
  related: [externalResource],
  rendering: [externalResource],
  otherContent: [annotationList],
  within: within,

  // Extra
  thumbnail: imageResource,
});

// ===========================================================================
// 3) Sequences
// ===========================================================================
sequence.define({
  canvases: [canvas],

  // Linking
  seeAlso: [externalResource],
  service: [service],
  related: [externalResource],
  rendering: [externalResource],
  otherContent: [annotationList],
  within: within,
  startCanvas: canvas,

  // Extra
  thumbnail: imageResource,
});

// ===========================================================================
// 4) Canvases
// ===========================================================================
canvas.define({
  images: [annotation],

  // Linking
  seeAlso: [externalResource],
  service: [service],
  related: [externalResource],
  rendering: [externalResource],
  otherContent: [annotationList],
  within: within,

  // Extra
  thumbnail: imageResource,
});

// ===========================================================================
// 5) Annotation
// ===========================================================================
annotation.define({
  resource: new schema.Union(
    {
      imageResource,
      externalResource,
      choice,
    },
    input => {
      switch (input['@type']) {
        case 'dctypes:Image':
          return 'imageResource';
        case 'oa:Choice':
          return 'choice';
        default:
          return 'externalResource';
      }
    }
  ),
  on: canvas,

  // Linking
  seeAlso: [externalResource],
  service: [service],
  related: [externalResource],
  rendering: [externalResource],
  within: within,

  // Extra
  thumbnail: imageResource,
});

// ===========================================================================
// 6) Annotation List
// ===========================================================================
annotationList.define({
  resources: [annotation],

  // Linking
  seeAlso: [externalResource],
  service: [service],
  related: [externalResource],
  rendering: [externalResource],
  within: within,

  // Extra
  thumbnail: imageResource,
});

// ===========================================================================
// 7) Range
// ===========================================================================
range.define({
  resources: [annotation],
  members: [rangeMember],
  canvases: [canvas],
  ranges: [range],

  // Linking
  seeAlso: [externalResource],
  service: [service],
  related: [externalResource],
  rendering: [externalResource],
  within: within,
  startCanvas: canvas,
  contentLayer: layer,

  // Extra
  thumbnail: imageResource,
});

// ===========================================================================
// 8) Layer
// ===========================================================================
layer.define({
  // Linking
  seeAlso: [externalResource],
  service: [service],
  related: [externalResource],
  rendering: [externalResource],
  within: within,
});

// ===========================================================================
// 9) Image resource
// ===========================================================================
imageResource.define({
  service: [service],
});

// ===========================================================================
// 10) Choice
// ===========================================================================
choice.define({
  default: annotation,
  item: [annotation],
});

const preprocess = compose(
  moveStartCanvasToSequence,
  preprocessLinkedEntities,
  addMissingIds
);

const normalizeResource = (rawResource, customSchema = resource) =>
  normalize(preprocess(rawResource), customSchema);

export {
  normalizeResource as normalize,
  preprocess,
  resource,
  collection,
  manifest,
  sequence,
  canvas,
  annotation,
  annotationList,
  range,
  layer,
  imageResource,
  externalResource,
  within,
  service,
};
