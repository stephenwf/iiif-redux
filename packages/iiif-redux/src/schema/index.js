import * as presentation2 from './presentation2';
import * as presentation3 from './presentation3';

export { presentation2, presentation3 };

// Default is 2 until 3 is ratified?
export default presentation2;
