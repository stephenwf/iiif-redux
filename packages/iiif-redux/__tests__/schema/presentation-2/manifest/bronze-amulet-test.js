import bronzeAmulet from '../../../fixtures/bronze-amulet';
import { normalize } from '../../../../src/schema/presentation2';

describe('schema/presentation-2/manifest/bronze-amulet', () => {
  it('should parse manifest', () => {
    const entity = normalize(bronzeAmulet);
    expect(entity).toMatchSnapshot();
  });
});
