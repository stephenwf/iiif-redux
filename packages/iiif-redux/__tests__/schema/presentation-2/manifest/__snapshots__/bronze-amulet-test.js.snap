// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`schema/presentation-2/manifest/bronze-amulet should parse manifest 1`] = `
Object {
  "entities": Object {
    "annotationLists": Object {
      "http://jubilees.stmarytx.edu/iiifp/TAW-113/list/recto.json": Object {
        "@id": "http://jubilees.stmarytx.edu/iiifp/TAW-113/list/recto.json",
        "@type": "sc:AnnotationList",
      },
      "http://jubilees.stmarytx.edu/iiifp/TAW-113/list/verso.json": Object {
        "@id": "http://jubilees.stmarytx.edu/iiifp/TAW-113/list/verso.json",
        "@type": "sc:AnnotationList",
      },
    },
    "annotations": Object {
      "http://jubilees.stmarytx.edu/iiif/TAW-113-Amulet-r-Ac-00.jp2/full/full/0/default.jpg": Object {
        "@id": "http://jubilees.stmarytx.edu/iiif/TAW-113-Amulet-r-Ac-00.jp2/full/full/0/default.jpg",
        "@type": "dctypes:Image",
        "format": "image/jpeg",
        "height": 8181,
        "label": Array [
          Object {
            "@language": "en",
            "@value": "Accurate Color Diffuse",
          },
        ],
        "service": Array [
          "http://jubilees.stmarytx.edu/iiif/TAW-113-Amulet-r-Ac-00.jp2",
        ],
        "width": 6132,
      },
      "http://jubilees.stmarytx.edu/iiif/TAW-113-Amulet-r-Ac-02.jp2/full/full/0/default.jpg": Object {
        "@id": "http://jubilees.stmarytx.edu/iiif/TAW-113-Amulet-r-Ac-02.jp2/full/full/0/default.jpg",
        "@type": "dctypes:Image",
        "format": "image/jpeg",
        "height": 8181,
        "label": Array [
          Object {
            "@language": "en",
            "@value": "Accurate Color Raking 1",
          },
        ],
        "service": Array [
          "http://jubilees.stmarytx.edu/iiif/TAW-113-Amulet-r-Ac-02.jp2",
        ],
        "width": 6132,
      },
      "http://jubilees.stmarytx.edu/iiif/TAW-113-Amulet-r-Ac-07.jp2/full/full/0/default.jpg": Object {
        "@id": "http://jubilees.stmarytx.edu/iiif/TAW-113-Amulet-r-Ac-07.jp2/full/full/0/default.jpg",
        "@type": "dctypes:Image",
        "format": "image/jpeg",
        "height": 8181,
        "label": Array [
          Object {
            "@language": "en",
            "@value": "Accurate Color Raking 2",
          },
        ],
        "service": Array [
          "http://jubilees.stmarytx.edu/iiif/TAW-113-Amulet-r-Ac-07.jp2",
        ],
        "width": 6132,
      },
      "http://jubilees.stmarytx.edu/iiif/TAW-113-Amulet-r-Ac-51.jp2/full/full/0/default.jpg": Object {
        "@id": "http://jubilees.stmarytx.edu/iiif/TAW-113-Amulet-r-Ac-51.jp2/full/full/0/default.jpg",
        "@type": "dctypes:Image",
        "format": "image/jpeg",
        "height": 8181,
        "label": Array [
          Object {
            "@language": "en",
            "@value": "Accurate Color Raking 3",
          },
        ],
        "service": Array [
          "http://jubilees.stmarytx.edu/iiif/TAW-113-Amulet-r-Ac-51.jp2",
        ],
        "width": 6132,
      },
      "http://jubilees.stmarytx.edu/iiif/TAW-113-Amulet-r-Ac-55.jp2/full/full/0/default.jpg": Object {
        "@id": "http://jubilees.stmarytx.edu/iiif/TAW-113-Amulet-r-Ac-55.jp2/full/full/0/default.jpg",
        "@type": "dctypes:Image",
        "format": "image/jpeg",
        "height": 8181,
        "label": Array [
          Object {
            "@language": "en",
            "@value": "Accurate Color Raking 4",
          },
        ],
        "service": Array [
          "http://jubilees.stmarytx.edu/iiif/TAW-113-Amulet-r-Ac-55.jp2",
        ],
        "width": 6132,
      },
      "http://jubilees.stmarytx.edu/iiif/TAW-113-Amulet-v-Ac-00.jp2/full/full/0/default.jpg": Object {
        "@id": "http://jubilees.stmarytx.edu/iiif/TAW-113-Amulet-v-Ac-00.jp2/full/full/0/default.jpg",
        "@type": "dctypes:Image",
        "format": "image/jpeg",
        "height": 7216,
        "label": Array [
          Object {
            "@language": "en",
            "@value": "Accurate Color Diffuse",
          },
        ],
        "service": Array [
          "http://jubilees.stmarytx.edu/iiif/TAW-113-Amulet-v-Ac-00.jp2",
        ],
        "width": 5412,
      },
      "http://jubilees.stmarytx.edu/iiif/TAW-113-Amulet-v-Ac-02.jp2/full/full/0/default.jpg": Object {
        "@id": "http://jubilees.stmarytx.edu/iiif/TAW-113-Amulet-v-Ac-02.jp2/full/full/0/default.jpg",
        "@type": "dctypes:Image",
        "format": "image/jpeg",
        "height": 7216,
        "label": Array [
          Object {
            "@language": "en",
            "@value": "Accurate Color Raking 1",
          },
        ],
        "service": Array [
          "http://jubilees.stmarytx.edu/iiif/TAW-113-Amulet-v-Ac-02.jp2",
        ],
        "width": 5412,
      },
      "http://jubilees.stmarytx.edu/iiif/TAW-113-Amulet-v-Ac-07.jp2/full/full/0/default.jpg": Object {
        "@id": "http://jubilees.stmarytx.edu/iiif/TAW-113-Amulet-v-Ac-07.jp2/full/full/0/default.jpg",
        "@type": "dctypes:Image",
        "format": "image/jpeg",
        "height": 7216,
        "label": Array [
          Object {
            "@language": "en",
            "@value": "Accurate Color Raking 2",
          },
        ],
        "service": Array [
          "http://jubilees.stmarytx.edu/iiif/TAW-113-Amulet-v-Ac-07.jp2",
        ],
        "width": 5412,
      },
      "http://jubilees.stmarytx.edu/iiif/TAW-113-Amulet-v-Ac-51.jp2/full/full/0/default.jpg": Object {
        "@id": "http://jubilees.stmarytx.edu/iiif/TAW-113-Amulet-v-Ac-51.jp2/full/full/0/default.jpg",
        "@type": "dctypes:Image",
        "format": "image/jpeg",
        "height": 7216,
        "label": Array [
          Object {
            "@language": "en",
            "@value": "Accurate Color Raking 3",
          },
        ],
        "service": Array [
          "http://jubilees.stmarytx.edu/iiif/TAW-113-Amulet-v-Ac-51.jp2",
        ],
        "width": 5412,
      },
      "http://jubilees.stmarytx.edu/iiif/TAW-113-Amulet-v-Ac-55.jp2/full/full/0/default.jpg": Object {
        "@id": "http://jubilees.stmarytx.edu/iiif/TAW-113-Amulet-v-Ac-55.jp2/full/full/0/default.jpg",
        "@type": "dctypes:Image",
        "format": "image/jpeg",
        "height": 7216,
        "label": Array [
          Object {
            "@language": "en",
            "@value": "Accurate Color Raking 4",
          },
        ],
        "service": Array [
          "http://jubilees.stmarytx.edu/iiif/TAW-113-Amulet-v-Ac-55.jp2",
        ],
        "width": 5412,
      },
      "https://975b3341312ee6360d90ff9f623da6e040babcdd": Object {
        "@id": "https://975b3341312ee6360d90ff9f623da6e040babcdd",
        "@type": "oa:Annotation",
        "motivation": "sc:painting",
        "on": "http://jubilees.stmarytx.edu/iiifp/TAW-113/canvas/Amulet-Recto",
        "resource": Object {
          "id": "http://jubilees.stmarytx.edu/iiif/TAW-113-Amulet-r-Ac-00.jp2/full/full/0/default.jpg",
          "schema": "choice",
        },
      },
      "https://f03efc7dcfeb75acc48a1ad0b63451eff8e6b7de": Object {
        "@id": "https://f03efc7dcfeb75acc48a1ad0b63451eff8e6b7de",
        "@type": "oa:Annotation",
        "motivation": "sc:painting",
        "on": "http://jubilees.stmarytx.edu/iiifp/TAW-113/canvas/Amulet-Verso",
        "resource": Object {
          "id": "http://jubilees.stmarytx.edu/iiif/TAW-113-Amulet-v-Ac-00.jp2/full/full/0/default.jpg",
          "schema": "choice",
        },
      },
    },
    "canvases": Object {
      "http://jubilees.stmarytx.edu/iiifp/TAW-113/canvas/Amulet-Recto": Object {
        "@context": "http://iiif.io/api/presentation/2/context.json",
        "@id": "http://jubilees.stmarytx.edu/iiifp/TAW-113/canvas/Amulet-Recto",
        "@type": "sc:Canvas",
        "height": 8176,
        "images": Array [
          "https://975b3341312ee6360d90ff9f623da6e040babcdd",
        ],
        "label": Array [
          Object {
            "@language": "en",
            "@value": "Amulet-Recto",
          },
        ],
        "otherContent": Array [
          "http://jubilees.stmarytx.edu/iiifp/TAW-113/list/recto.json",
        ],
        "thumbnail": "http://jubilees.stmarytx.edu/iiif/TAW-113-Amulet-r-Ac-00.jp2/square/100,100/0/default.jpg",
        "width": 6132,
      },
      "http://jubilees.stmarytx.edu/iiifp/TAW-113/canvas/Amulet-Verso": Object {
        "@context": "http://iiif.io/api/presentation/2/context.json",
        "@id": "http://jubilees.stmarytx.edu/iiifp/TAW-113/canvas/Amulet-Verso",
        "@type": "sc:Canvas",
        "height": 7216,
        "images": Array [
          "https://f03efc7dcfeb75acc48a1ad0b63451eff8e6b7de",
        ],
        "label": Array [
          Object {
            "@language": "en",
            "@value": "Amulet-Verso",
          },
        ],
        "otherContent": Array [
          "http://jubilees.stmarytx.edu/iiifp/TAW-113/list/verso.json",
        ],
        "thumbnail": "http://jubilees.stmarytx.edu/iiif/TAW-113-Amulet-v-Ac-00.jp2/square/100,100/0/default.jpg",
        "width": 5412,
      },
    },
    "choices": Object {
      "http://jubilees.stmarytx.edu/iiif/TAW-113-Amulet-r-Ac-00.jp2/full/full/0/default.jpg": Object {
        "@id": "http://jubilees.stmarytx.edu/iiif/TAW-113-Amulet-r-Ac-00.jp2/full/full/0/default.jpg",
        "@type": "oa:Choice",
        "default": "http://jubilees.stmarytx.edu/iiif/TAW-113-Amulet-r-Ac-00.jp2/full/full/0/default.jpg",
        "item": Array [
          "http://jubilees.stmarytx.edu/iiif/TAW-113-Amulet-r-Ac-00.jp2/full/full/0/default.jpg",
          "http://jubilees.stmarytx.edu/iiif/TAW-113-Amulet-r-Ac-02.jp2/full/full/0/default.jpg",
          "http://jubilees.stmarytx.edu/iiif/TAW-113-Amulet-r-Ac-07.jp2/full/full/0/default.jpg",
          "http://jubilees.stmarytx.edu/iiif/TAW-113-Amulet-r-Ac-51.jp2/full/full/0/default.jpg",
          "http://jubilees.stmarytx.edu/iiif/TAW-113-Amulet-r-Ac-55.jp2/full/full/0/default.jpg",
        ],
      },
      "http://jubilees.stmarytx.edu/iiif/TAW-113-Amulet-v-Ac-00.jp2/full/full/0/default.jpg": Object {
        "@id": "http://jubilees.stmarytx.edu/iiif/TAW-113-Amulet-v-Ac-00.jp2/full/full/0/default.jpg",
        "@type": "oa:Choice",
        "default": "http://jubilees.stmarytx.edu/iiif/TAW-113-Amulet-v-Ac-00.jp2/full/full/0/default.jpg",
        "item": Array [
          "http://jubilees.stmarytx.edu/iiif/TAW-113-Amulet-v-Ac-00.jp2/full/full/0/default.jpg",
          "http://jubilees.stmarytx.edu/iiif/TAW-113-Amulet-v-Ac-02.jp2/full/full/0/default.jpg",
          "http://jubilees.stmarytx.edu/iiif/TAW-113-Amulet-v-Ac-07.jp2/full/full/0/default.jpg",
          "http://jubilees.stmarytx.edu/iiif/TAW-113-Amulet-v-Ac-51.jp2/full/full/0/default.jpg",
          "http://jubilees.stmarytx.edu/iiif/TAW-113-Amulet-v-Ac-55.jp2/full/full/0/default.jpg",
        ],
      },
    },
    "externalResources": Object {
      "http://jubilees.stmarytx.edu/index.html": Object {
        "@id": "http://jubilees.stmarytx.edu/index.html",
        "@type": "dctypes:text",
        "format": "text/html",
      },
    },
    "imageResources": Object {
      "http://jubilees.stmarytx.edu/iiif/TAW-113-Amulet-r-Ac-00.jp2/square/100,100/0/default.jpg": Object {
        "@id": "http://jubilees.stmarytx.edu/iiif/TAW-113-Amulet-r-Ac-00.jp2/square/100,100/0/default.jpg",
        "@type": "dctypes:Image",
        "format": "image/jpeg",
        "service": Array [
          "http://jubilees.stmarytx.edu/iiif/TAW-113-Amulet-r-Ac-00.jp2",
        ],
      },
      "http://jubilees.stmarytx.edu/iiif/TAW-113-Amulet-v-Ac-00.jp2/square/100,100/0/default.jpg": Object {
        "@id": "http://jubilees.stmarytx.edu/iiif/TAW-113-Amulet-v-Ac-00.jp2/square/100,100/0/default.jpg",
        "@type": "dctypes:Image",
        "format": "image/jpeg",
        "service": Array [
          "http://jubilees.stmarytx.edu/iiif/TAW-113-Amulet-v-Ac-00.jp2",
        ],
      },
    },
    "manifests": Object {
      "http://jubilees.stmarytx.edu/iiifp/TAW-113/manifest.json": Object {
        "@context": "http://iiif.io/api/presentation/2/context.json",
        "@id": "http://jubilees.stmarytx.edu/iiifp/TAW-113/manifest.json",
        "@type": "sc:Manifest",
        "attribution": Array [
          Object {
            "@language": "en",
            "@value": "Jubilees Palimpsest Project, TAW",
          },
        ],
        "description": Array [
          Object {
            "@language": "en",
            "@value": "Bronze Amulet, 3rd-4th Century CE, Greek Inscriptions of Various Hebrew and Greek Names of God",
          },
        ],
        "label": Array [
          Object {
            "@language": "en",
            "@value": "Bronze Amulet",
          },
        ],
        "license": "https://creativecommons.org/licenses/by-nc-sa/4.0/",
        "logo": Object {
          "@id": "http://jubilees.stmarytx.edu/iiif/Logo-JubPalProj.jp2/full/full/0/default.jpg",
          "@type": "dctypes:Image",
          "service": Object {
            "@context": "http://iiif.io/api/image/2/context.json",
            "@id": "http://jubilees.stmarytx.edu/iiif/Logo-JubPalProj.jp2",
            "profile": "http://iiif.io/api/image/2/profiles/level2.json",
          },
        },
        "metadata": Array [
          Object {
            "label": Array [
              Object {
                "@language": "en",
                "@value": "Author",
              },
            ],
            "value": Array [
              Object {
                "@language": "en",
                "@value": "Jubilees Palimpest Project",
              },
            ],
          },
          Object {
            "label": Array [
              Object {
                "@language": "en",
                "@value": "Published",
              },
            ],
            "value": Array [
              Object {
                "@language": "en",
                "@value": "2016",
              },
            ],
          },
          Object {
            "label": Array [
              Object {
                "@language": "en",
                "@value": "Source",
              },
            ],
            "value": Array [
              Object {
                "@language": "en",
                "@value": "http://jubilees.stmarytx.edu",
              },
            ],
          },
        ],
        "navDate": "300-01-01T00:00:00Z",
        "related": Array [
          "http://jubilees.stmarytx.edu/index.html",
        ],
        "sequences": Array [
          "http://jubilees.stmarytx.edu/iiifp/TAW-113/sequence/chronological",
        ],
        "thumbnail": "http://jubilees.stmarytx.edu/iiif/TAW-113-Amulet-r-Ac-00.jp2/square/100,100/0/default.jpg",
        "viewingDirection": "left-to-right",
        "viewingHint": "paged",
      },
    },
    "sequences": Object {
      "http://jubilees.stmarytx.edu/iiifp/TAW-113/sequence/chronological": Object {
        "@context": "http://iiif.io/api/presentation/2/context.json",
        "@id": "http://jubilees.stmarytx.edu/iiifp/TAW-113/sequence/chronological",
        "@type": "sc:Sequence",
        "canvases": Array [
          "http://jubilees.stmarytx.edu/iiifp/TAW-113/canvas/Amulet-Recto",
          "http://jubilees.stmarytx.edu/iiifp/TAW-113/canvas/Amulet-Verso",
        ],
        "label": Array [
          Object {
            "@language": "en",
            "@value": "Recto to Verso",
          },
        ],
        "startCanvas": "http://jubilees.stmarytx.edu/iiifp/TAW-113/canvas/Amulet-Recto",
        "viewingDirection": "left-to-right",
        "viewingHint": "paged",
      },
    },
    "services": Object {
      "http://jubilees.stmarytx.edu/iiif/TAW-113-Amulet-r-Ac-00.jp2": Object {
        "@context": "http://iiif.io/api/image/2/context.json",
        "@id": "http://jubilees.stmarytx.edu/iiif/TAW-113-Amulet-r-Ac-00.jp2",
        "profile": "http://iiif.io/api/image/2/level1.json",
      },
      "http://jubilees.stmarytx.edu/iiif/TAW-113-Amulet-r-Ac-02.jp2": Object {
        "@context": "http://iiif.io/api/image/2/context.json",
        "@id": "http://jubilees.stmarytx.edu/iiif/TAW-113-Amulet-r-Ac-02.jp2",
        "profile": "http://iiif.io/api/image/2/level1.json",
      },
      "http://jubilees.stmarytx.edu/iiif/TAW-113-Amulet-r-Ac-07.jp2": Object {
        "@context": "http://iiif.io/api/image/2/context.json",
        "@id": "http://jubilees.stmarytx.edu/iiif/TAW-113-Amulet-r-Ac-07.jp2",
        "profile": "http://iiif.io/api/image/2/level1.json",
      },
      "http://jubilees.stmarytx.edu/iiif/TAW-113-Amulet-r-Ac-51.jp2": Object {
        "@context": "http://iiif.io/api/image/2/context.json",
        "@id": "http://jubilees.stmarytx.edu/iiif/TAW-113-Amulet-r-Ac-51.jp2",
        "profile": "http://iiif.io/api/image/2/level1.json",
      },
      "http://jubilees.stmarytx.edu/iiif/TAW-113-Amulet-r-Ac-55.jp2": Object {
        "@context": "http://iiif.io/api/image/2/context.json",
        "@id": "http://jubilees.stmarytx.edu/iiif/TAW-113-Amulet-r-Ac-55.jp2",
        "profile": "http://iiif.io/api/image/2/level1.json",
      },
      "http://jubilees.stmarytx.edu/iiif/TAW-113-Amulet-v-Ac-00.jp2": Object {
        "@context": "http://iiif.io/api/image/2/context.json",
        "@id": "http://jubilees.stmarytx.edu/iiif/TAW-113-Amulet-v-Ac-00.jp2",
        "profile": "http://iiif.io/api/image/2/level1.json",
      },
      "http://jubilees.stmarytx.edu/iiif/TAW-113-Amulet-v-Ac-02.jp2": Object {
        "@context": "http://iiif.io/api/image/2/context.json",
        "@id": "http://jubilees.stmarytx.edu/iiif/TAW-113-Amulet-v-Ac-02.jp2",
        "profile": "http://iiif.io/api/image/2/level1.json",
      },
      "http://jubilees.stmarytx.edu/iiif/TAW-113-Amulet-v-Ac-07.jp2": Object {
        "@context": "http://iiif.io/api/image/2/context.json",
        "@id": "http://jubilees.stmarytx.edu/iiif/TAW-113-Amulet-v-Ac-07.jp2",
        "profile": "http://iiif.io/api/image/2/level1.json",
      },
      "http://jubilees.stmarytx.edu/iiif/TAW-113-Amulet-v-Ac-51.jp2": Object {
        "@context": "http://iiif.io/api/image/2/context.json",
        "@id": "http://jubilees.stmarytx.edu/iiif/TAW-113-Amulet-v-Ac-51.jp2",
        "profile": "http://iiif.io/api/image/2/level1.json",
      },
      "http://jubilees.stmarytx.edu/iiif/TAW-113-Amulet-v-Ac-55.jp2": Object {
        "@context": "http://iiif.io/api/image/2/context.json",
        "@id": "http://jubilees.stmarytx.edu/iiif/TAW-113-Amulet-v-Ac-55.jp2",
        "profile": "http://iiif.io/api/image/2/level1.json",
      },
    },
  },
  "result": Object {
    "id": "http://jubilees.stmarytx.edu/iiifp/TAW-113/manifest.json",
    "schema": "manifest",
  },
}
`;
